react/InfosCarrousel/index.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import SwipeableViews from 'react-swipeable-views'
+
+import styles from './styles.styl'
+import Icon from '../Icon'
+
+const InfosCarrousel = ({ children, theme, className, swipleableProps }) => {
+  const count = React.Children.count(children)
+  const [index, setIndex] = useState(0)
+  const nextInfos = useCallback(() => setIndex(index + 1), [index])
+  const previousInfos = useCallback(() => setIndex(index - 1), [index])
+  const hasPreviousInfos = index === 0
+  const hasNextInfos = index === count - 1
+
+  return (
+    <div
+      className={cx(
+        styles['InfosCarrousel'],
+        styles[`InfosCarrousel--${theme}`],
+        className
+      )}
+    >
+      <SwipeableViews index={index} animateHeight {...swipleableProps}>
+        {children}
```

Comment:
To remove the need to repeat the theme property on the children, we can do `React.Children.map(children x => React.clone(x, { theme }))`

Reply:
I thought about that, but it feels like a bad practice to forcefully inject props to child components… or is it just me?

Reply:
I think there a times when it is necessary, mainly when the coupling between the parent and the child is very visible, and expected.

I use the same strategy for example with the Slides inside a Swiper component. They receive props like isActiveSlide.

> it feels like a bad practice to __forcefully__ inject props to child components

A check not to override an existing theme prop would be good I think.

Reply:
Another possible technique would be to use a themeContext, a provider inside themeInfos, an a consumer inside Infos.

```
const Infos = () => {
  let contextTheme = useContext(ThemeContext)
  let resolvedTheme = props.theme || contextTheme
   ....
}
```

Reply:
Another way to implement this `context` need is to use the cascading styles of CSS.

```
.themePrimary .InfosSlide
  background var(--primaryColor)

.themePrimary .Infos
  background var(--primaryColor)
```

Reply:
Cascading css isn't possible I think because of css modules — `Infos` and `InfosCarrousel` are 2 separate modules. The context would work, but I'll go ahead with the `clone` technique, checking that there is no theme prop we would override 👍

Reply:
In this case, I wonder if Infos and InfosCarrousel should not share the same stylus module. Anyway, this goes beyond the scope of the PR.

> with the clone technique, checking that there is no theme prop we would override 👍

+1